src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java

```diff
@@ -60,7 +60,11 @@ public void connect(String address, int port) throws IOException {
             logger.info("Connected successfully to {}", address);
         } catch (Exception e) {
             logger.error("Could not connect to FTP server '{}'. {}.", address, getStatusString());
-            this.disconnect();
+            try {
+                this.disconnect();
+            } catch (Exception ex) {
```

Comment:
Is it not the same Exception as in `NCBIAssemblyDataSource` ?

Reply:
It is, just getting thrown at a different place. I am handling this wherever we are calling disconnect method because I didn't want to change the behaviour of disconnect method by handling it inside it.

> Do we need the same try/catch in [ENAAssemblyDataSource](https://github.com/EBIvariation/contig-alias/blob/16c4ee6c56a3219eba384b5fe15529380b417ac7/src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java#L73) as well?

Good catch April. We can handle it here as well, updated the code.

Reply:
Why are we catching `Exception` and not `NCBIAssemblyDataSource` then ?

Reply:
updated to catch IOException